benchmark/atoi.cpp

```diff
@@ -256,13 +276,18 @@ Return c_strToIntegral(const char *str) noexcept {
 }
 
 int c_strToI(const char *str) noexcept {
-    return impl::c_strToIntegral<int>(str);
+    return impl::c_strToIntegral<int, calculateBase10>(str);
```

Comment:
From reading this line of code, it gave me a thought and I'd like to see if you guys think it has any merit. 

Does it make sense to allow people to provide signed types in the interface to SWARs and the like? The signedness has semantic meaning which we disregard since we only really are caring about the total numbers of bits available in that type. 

Would it be meaningful to SFINAE the assertion of unsigned types to the API of relevant places to effectively communicate that to users? 

Not sure if huge value in this, but an idea I wanted to see if you guys thought there was any merit to the idea.

Reply:
The reason we need the base type to be unsigned is the need for shift rights to be logical and not arithmetic:
In arithmetic shift rights ("signed" shift right) the most significant bit is copied down.  From the SWAR point of view, this behavior affects only the most significant lane.
I will create an issue to assign it to you to implement Lane-wise arithmetic shift right.
I have been clear about this from the beginning, after Robin Hood, when we bgan working again on the library, it is when I decided that we can not really support signed base types, and thus we may as well let users specify a signed type as a base to use its unsigned form.  The deficiency of this approach is that `SWAR<8, int>` and `SWAR<8, unsigned>` are different types.
We need to separate the user-facing `SWAR` from the implementation, but I am still not sure of how to do this; I mean I do not have all the answers clear to just do it.